### ISSUE TITLE
Apply color on image template

### ISSUE THREAD

**Issue by aleufms** [Raizlabs/BonMot#105]
```swift
let chain = BONChain().textColor(UIColor.redColor())
chain.appendLink(BONChain().image(UIImage(named: "image_name")))
chain.appendLink(BONChain().string("TEXT").text, separator: " ")
chain.appendLink(BONChain().image(UIImage(named: "image_name")), separator: " ")
```

The red color only is applied on the lastimage. The first image on the chain remains black.

**Comment by ZevEisenberg** [comment]
@aleufms thanks for your report! Would you be interested in submitting a pull request to fix it? Writing a unit tests that demonstrates the failure is a good first step, although in this case, determining that the resulting string contains tinted images could be tricky. Let me know if you'd like to try - I'm more than happy to help out if you'd like.

**Comment by ZevEisenberg** [comment]
Come to think of it, I'm a little confused by what you're seeing. BonMot doesn't currently automatically tint images based on their text color, at least not until someone fixes #21. Could you provide a little more information, and maybe a sample image, of what you're seeing, @aleufms?

**Comment by aleufms** [comment]
Ok, sorry for the delay. I've been upload an example on https://github.com/aleufms/BonMotImageExample

**Comment by ZevEisenberg** [comment]
TL;DR you found a bug or secret feature in UIKit. Scroll to the bottom for my recommendation of what yo should do.

OK, so, you've stumbled upon what is, as far as I can tell, either a bug or an undocumented feature of `NSAttributedString`:

```swift
let attachment = NSTextAttachment()

// image must be set to Template rendering mode
attachment.image = UIImage(named: "ic_navigate_next")

// Need to use a space, or this bug/feature isn't activated
let attributedString = NSMutableAttributedString(string: " ")

let attachmentString = NSAttributedString(attachment: attachment)

attributedString.appendAttributedString(attachmentString)
attributedString.addAttribute(NSForegroundColorAttributeName, value: UIColor.orangeColor(), range: NSMakeRange(0, 2))
label.attributedText = attributedString
```

This will result in a colored image:

<img width="49" alt="screen shot 2016-02-09 at 9 57 31 pm" src="https://cloud.githubusercontent.com/assets/464574/12937593/22d19df8-cf78-11e5-995a-1aff6e62b2fc.png">

(The original image was black.)

If you do just an image, without a space, you don't get color:

```swift
let attachment = NSTextAttachment()
attachment.image = UIImage(named: "ic_navigate_next")

let attachmentString = NSAttributedString(attachment: attachment).mutableCopy() as! NSMutableAttributedString

attachmentString.addAttribute(NSForegroundColorAttributeName, value: UIColor.orangeColor(), range: NSMakeRange(0, 1))
label.attributedText = attachmentString
```

<img width="57" alt="screen shot 2016-02-09 at 10 01 16 pm" src="https://cloud.githubusercontent.com/assets/464574/12937640/ad6a7c28-cf78-11e5-8540-ccb6bd2bf273.png">

I didn't know about this behavior until you posted this bug, and it's not supported in BonMot - see #21. I wonder if there's a way, in BonMot, to _prevent_ it, so people are not confused in the future? For example, we could force the image to always render in Original mode in `-attributedStringLastConcatenant:` like this:

```objc
- (NSAttributedString *)attributedStringLastConcatenant:(BOOL)lastConcatenant
{
    NSMutableAttributedString *mutableAttributedString = nil;

    if (self.image) {
        NSAssert(!self.string, @"If self.image is non-nil, self.string must be nil");
        NSTextAttachment *attachment = [[NSTextAttachment alloc] init];
        attachment.image = [self.image imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
```

@aleufms for your use case, I recommend pre-tinting your image using `-[UIImage bon_tintedImageWithColor:]` from `UIImage+BonMotUtilities`.

**Comment by ZevEisenberg** [comment]
Acceptance criteria for BonMot 4.0:

- [ ] Exercise bug somewhere verifiable (demo app or unit test)
- [ ] Decide on re-tinting image or non-breaking space as method of fixing.
- [ ] Implement fix

**Comment by KingOfBrian** [comment]
While commenting the method, I thought that we might want to support background color. Does that make sense to you?

**Comment by ZevEisenberg** [comment]
Background color for the image? Wouldn't NSBackgroundColorAttributeName take care of that?

**Comment by KingOfBrian** [comment]
Yea, not sure what I was thinking there. Carry on!

**Comment by ZevEisenberg** [comment]
In the upcoming 4.0 release, images set to "automatic" or "always template" will now be tinted if they are styled with a color.
